content/article/softlayer_search.md

```diff
@@ -0,0 +1,224 @@
+---
+title: "Searching with the SoftLayer API"
+description: "An explanation of how to use the SoftLayer_Search service."
+date: "2021-06-18"
+tags:
+    - "article"
+    - "sldn"
+---
+
+# Introduction
+
+>The SoftLayer_Search service is only available in v3.1 of the API. You will get a "Service does not exist" error if you use v3
+
+The [SoftLayer_Search](/reference/services/SoftLayer_Search/) is an API service that lets you make complex queries about data indexed by the service. While most of the SoftLayer API is contained in a SQL database, the SoftLayer_Search service uses an Elastic Search database to generate its results, allowing for more complex and responsive queries.
+
+To start off, we need to understand what type of objects SoftLayer_Search knows about, and do to this, we use the [SoftLayer_Search::getObjectTypes()](/reference/services/SoftLayer_Search/getObjectTypes/) API Call.
+
+## Searchable Types
+As of this writing (2021)
+
+- SoftLayer_Event_Log
+- SoftLayer_Virtual_DedicatedHost
+- SoftLayer_Hardware
+- SoftLayer_Network_Application_Delivery_Controller
+- SoftLayer_Network_Subnet_IpAddress
+- SoftLayer_Network_Vlan
+- SoftLayer_Network_Vlan_Firewall
+- SoftLayer_Ticket
+- SoftLayer_Virtual_Guest
+
+
+# [SoftLayer_Search::getObjectTypes](#getObjectTypes) {#getObjectTypes .anchor-link}
+
+The API call here is simple enough.
+
+
+```bash
+curl -u $SL_USER:$SL_APIKEY 'https://api.softlayer.com/rest/v3.1/SoftLayer_Search/getObjectTypes.json'
+```
+
+```bash
+slcli --format=json call-api SoftLayer_Search getObjectTypes
+```
+
+```
+ibmcloud sl call-api call-api SoftLayer_Search getObjectTypes
+```
+
+The output will look something like this:
+
+```json
+{
+    "name": "SoftLayer_Virtual_Guest",
+    "properties": [
+        {
+            "name": "accountId",
+            "sortableFlag": true,
+            "type": "integer"
+        },
+        {
+            "name": "datacenter.longName",
+            "sortableFlag": true,
+            "type": "string"
+        },
+```
+
+The `name` property referes to the type of SoftLayer Datatype being tracked, and the `properties` are the searchable fields, which are relational properties on the named datatype.
+
+# [SoftLayer_Search::search](#search_search) {#search_search .anchor-link}
+
+[SoftLayer_Search::search(search_term)](/reference/services/SoftLayer_Search/search/) takes in one parameter, and that is a string representing what you want to search for. The documentation here has a nice rundown of how to build out the API call. Basically you need to define which object types you want to search in, and what terms you want to search in those types for.
+
+## Find all Virtual and Hardware servers.
+
+By specifying an `_objectType` of `SoftLayer_Hardware` and ``SoftLayer_Virtual_Guest` and not defining a search term, this API call will return all Hardware and Virtual objects on your account.

```

Comment:
```suggestion
By specifying an `_objectType` of `SoftLayer_Hardware` and `SoftLayer_Virtual_Guest` and not defining a search term, this API call will return all Hardware and Virtual objects on your account.

```
looks like the is an additional single quote mark on `SoftLayer_Virtual_Guest`

Reply:
Thanks for catching that.